front/lib/resources/tracker_resource.ts

```diff
@@ -332,6 +334,43 @@ export class TrackerConfigurationResource extends ResourceWithSpace<TrackerConfi
     }));
   }
 
+  static async internalFetchAllWatchedForDocument(
+    auth: Authenticator,
+    dataSourceId: string,
+    parentIds: string[] | null
+  ): Promise<TrackerConfigurationResource[]> {
+    const dataSourceModelId = getResourceIdFromSId(dataSourceId);
+
+    if (!dataSourceModelId) {
+      throw new Error(`Invalid data source ID: ${dataSourceId}`);
+    }
+
+    const dsConfigs = await TrackerDataSourceConfigurationModel.findAll({
```

Comment:
NIT: could fetch only the attribute trackerConfigurationId?  Also, since in the end it uses auth and baseFetch to sure this needs to be called internal?

Reply:
Indeed